hooks/gate-scripts/lib/commit_scope.py

```
@@ -0,0 +1,359 @@
+# NOT -a/--all/--amend/--include/--only/--patch, not a bare operand (a pathspec),
+# not --no-verify. A message is REQUIRED: bare `git commit` opens core.editor.
+_FLAGS_NOARG = frozenset(("-q", "--quiet"))
+_FLAGS_ARG = frozenset(("-m", "--message"))
+
+_REGULAR_MODES = frozenset(("000000", "100644", "100755"))
+
+# See the bound in main(): the caller's per-path work must fit a 10s hook timeout.
+_MAX_PATHS = 20
+
+
+def _scan_ok(cmd):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_scan_ok`

Add return type annotation: `bool`

(ANN202)